### PR TITLE
refactor(cli): move common functions to `util.go`

### PR DESCRIPTION
**Pull request by shuangkun**
As https://github.com/argoproj/argo-workflows/pull/12803#issuecomment-2016698490  mentioned, I move "`unmarshal*` and `print*`" to `util.go`.

### CODE REVIEW

File: cmd/argo/commands/clustertemplate/list.go [hunk 1]
```diff
@@ -57,14 +54,3 @@ func NewListCommand() *cobra.Command {
 	command.Flags().StringVarP(&listArgs.output, "output", "o", "", "Output format. One of: wide|name")
 	return command
 }
-
-func printTable(wfList []wfv1.ClusterWorkflowTemplate) {
```
Comment (agilgur5) [inline cmd/argo/commands/clustertemplate/list.go:61]: `printTable` is actually _only_ used inside of the respective `list.go` files, so it shouldn't be moved. 

If you did a regex search on `print*`, that's my bad, I didn't realize this `printTable` function would be included in that search.
I only [meant](https://github.com/argoproj/argo-workflows/pull/12803#discussion_r1536717614) the `print<Resource>` functions, like `printClusterWorkflowTemplate*` (and the ones for `cron` and `template` commands, respectively).
`print<Resource>` and `unmarshal<Resource>` are used in multiple files, so they should really be defined in a common helper file like `util.go`. `printTable` is only used in one file, so it doesn't match them.

File: cmd/argo/commands/cron/list.go [hunk 2]
```diff
@@ -60,32 +55,3 @@ func NewListCommand() *cobra.Command {
 	command.Flags().StringVarP(&listArgs.labelSelector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2). Matching objects must satisfy all of the specified label constraints.")
 	return command
 }
-
-func printTable(wfList []wfv1.CronWorkflow, listArgs *listFlags) {
```
Comment (agilgur5) [inline cmd/argo/commands/cron/list.go:64]: not `printTable`, same as above

File: cmd/argo/commands/cron/util.go
Comment (agilgur5) [inline cmd/argo/commands/cron/util.go:1]: you missed moving `unmarshalCronWorkflows` here

File: cmd/argo/commands/template/list.go [hunk 3]
```diff
@@ -56,20 +53,3 @@ func NewListCommand() *cobra.Command {
 	command.Flags().StringVarP(&listArgs.output, "output", "o", "", "Output format. One of: wide|name")
 	return command
 }
-
-func printTable(wfList []wfv1.WorkflowTemplate, listArgs *listFlags) {
```
Comment (agilgur5) [inline cmd/argo/commands/template/list.go:60]: not `printTable`, same as above

**Review by agilgur5 — COMMENTED** [review]
`printTable` doesn't need to be moved, sorry about the confusion -- see in-line comment.

Also `unmarshalCronWorkflows` was missed -- see in-line comment.

File: cmd/argo/commands/clustertemplate/list.go (hunk 1 above)
Comment (shuangkun) [reply cmd/argo/commands/clustertemplate/list.go]: OK, I modified it, thanks！

File: cmd/argo/commands/cron/util.go
Comment (shuangkun) [reply cmd/argo/commands/cron/util.go:1]: Thanks, I forgot it before, added it now.

**Review by agilgur5 — APPROVED** [review]
Thanks for following up on this!

**Pull request merged**